### ISSUE TITLE
cuda "feature" causing build/install failure although my package has no cuda dependency

### ISSUE THREAD

**Issue by BrenBarn** [conda/conda-build#4056]
### Actual Behavior

I created a meta.yaml that has some dependencies.  As far as I can tell, none of these require CUDA.  But building and/or installing the package results in an error:

```
UnsatisfiableError: The following specifications were found to be incompatible with each other:

Output in format: Requested package -> Available versionsThe following specifications were found to be incompatible with your system:

  - feature:/win-64::__cuda==11.0=0

Your installed version is: 11.0
```

I have been trying different things and sometimes I get the error during the build, other times only when I try to actually install the package.

### Expected Behavior

The package should install without caring about CUDA.

### Steps to Reproduce

My meta.yaml has:

```
requirements:
  build:
    - python {{ python }}
    - setuptools
  run:
    - python
    - appdirs
    - toml
    - jupyter_client
    - pyqt
    - qscintilla2
```

`{{ python }}` is defined in `conda_build_config.yaml` as 3.6, 3.7 or 3.8.  As far as I can tell, none of those packages depend on CUDA.  I can't tell how or why conda thinks CUDA is relevant to my package at all.  I looked at issues [here](https://github.com/conda/conda/issues/4942) and [here](https://github.com/conda/conda/issues/2361) describing how to track down dependencies, but nothing I found led to CUDA from any of the listed packages.  I saw [this related issue](https://github.com/conda/conda/issues/9934) but in that case it seems the person was trying to install pytorch, which does use CUDA, whereas my package doesn't use it.

I don't know how to debug this to find out why conda thinks that CUDA has any relevance to building my package.  Looking through all the files in the packages generated by conda-build, I see no references to CUDA, so why is it involved in the dependency resolution at all?

##### `Output of conda info`
```
     active environment : testpydeal
    active env location : C:\FakeProgs\Anaconda3\envs\testpydeal
            shell level : 1
       user config file : C:\Users\BrenBarn\.condarc
 populated config files : C:\Users\BrenBarn\.condarc
          conda version : 4.8.4
    conda-build version : 3.20.2
         python version : 3.6.12.final.0
       virtual packages : __cuda=11.0
       base environment : C:\FakeProgs\Anaconda3  (writable)
           channel URLs : https://repo.anaconda.com/pkgs/main/win-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/win-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://repo.anaconda.com/pkgs/msys2/win-64
                          https://repo.anaconda.com/pkgs/msys2/noarch
          package cache : C:\FakeProgs\Anaconda3\pkgs
                          C:\Users\BrenBarn\.conda\pkgs
                          C:\Users\BrenBarn\AppData\Local\conda\conda\pkgs
       envs directories : C:\FakeProgs\Anaconda3\envs
                          C:\Users\BrenBarn\.conda\envs
                          C:\Users\BrenBarn\AppData\Local\conda\conda\envs
               platform : win-64
             user-agent : conda/4.8.4 requests/2.22.0 CPython/3.6.12 Windows/7 Windows/6.1.7601
          administrator : False
             netrc file : None
           offline mode : False
```

**Comment by pingvar** [comment]
Hi! 
I got the same error but when I'm trying to create a new environment, can't see that it has any dependency to cuda either

**Comment by BrenBarn** [comment]
I found that in my test environment I had forgotten to add `conda-forge` as a channel.  `qscintilla2` isn't available in defaults, only in conda-forge.  When I added conda-forge, my package installs correctly.

But this is still some kind of bug.  If the problem is that it can't find qscintilla2 (or some other package not available in the available channels) then the error should be that it can't find qscintilla2, not that it can't find some cuda feature that the to-be-installed package doesn't use (and that, as the error message shows, I already have anyway).
